File: shared-atlas-dart/lib/src/interfaces/iuser.dart
class IUser {
  final String id;
  final String username;
  final String email;
  final String? avatar;
  final DateTime createdAt;
  final DateTime updatedAt;

  IUser({
    required this.id,
    required this.username,
    required this.email,
    this.avatar,
    required this.createdAt,
    required this.updatedAt,
  });

  factory IUser.fromJson(Map<String, dynamic> json) {
    return IUser(
      id: json['id'] as String,
      username: json['username'] as String,
      email: json['email'] as String,
      avatar: json['avatar'] as String?,
      createdAt: DateTime.parse(json['createdAt'] as String),
      updatedAt: DateTime.parse(json['updatedAt'] as String),
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'id': this.id,
      'username': this.username,
      'email': this.email,
      'avatar': this.avatar,
      'createdAt': this.createdAt.toIso8601String(),
      'updatedAt': this.updatedAt.toIso8601String(),
    };
  }
}

File: shared-atlas-rust/src/interfaces/iuser.rs
use serde::{Serialize, Deserialize};

/// Represents a User entity in the system for frontend consumption.
/// Contains public profile information. Sensitive data like passwords are excluded.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct IUser {
/// Unique UUID of the user.
    pub id: String,
/// Unique username chosen by the user.
    pub username: String,
/// Email address of the user.
    pub email: String,
/// URL to the user's avatar image.
/// Can be a GCS signed URL or external URL (e.g., Google profile picture).
    pub avatar: Option<String>,
/// Timestamp when the user account was registered.
    pub created_at: chrono::DateTime<chrono::Utc>,
/// Timestamp when the user profile was last updated.
    pub updated_at: chrono::DateTime<chrono::Utc>,
}